traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1421,230 +1365,200 @@ func Update(w http.ResponseWriter, r *http.Request) {
 	}
 
-	userErr, sysErr, statusCode := api.CheckIfUnModified(r.Header, inf.Tx, *server.ID, "server")
+	userErr, sysErr, statusCode := api.CheckIfUnModified(inf.RequestHeaders(), inf.Tx, server.ID, "server")
 	if userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
-		return
+		return statusCode, userErr, sysErr
 	}
 
-	if server.CDNName != nil {
-		userErr, sysErr, statusCode = dbhelpers.CheckIfCurrentUserCanModifyCDN(inf.Tx.Tx, *server.CDNName, inf.User.UserName)
+	if server.CDN != "" {
+		userErr, sysErr, statusCode = dbhelpers.CheckIfCurrentUserCanModifyCDN(inf.Tx.Tx, server.CDN, inf.User.UserName)
 		if userErr != nil || sysErr != nil {
-			api.HandleErr(w, r, inf.Tx.Tx, statusCode, userErr, sysErr)
-			return
+			return statusCode, userErr, sysErr
 		}
-	} else if server.CDNID != nil {
-		userErr, sysErr, statusCode = dbhelpers.CheckIfCurrentUserCanModifyCDNWithID(inf.Tx.Tx, int64(*server.CDNID), inf.User.UserName)
+	} else {
+		userErr, sysErr, statusCode = dbhelpers.CheckIfCurrentUserCanModifyCDNWithID(inf.Tx.Tx, int64(server.CDNID), inf.User.UserName)
 		if userErr != nil || sysErr != nil {
-			api.HandleErr(w, r, inf.Tx.Tx, statusCode, userErr, sysErr)
-			return
+			return statusCode, userErr, sysErr
 		}
 	}
 
 	if inf.Version.Major >= 4 {
-		if err = dbhelpers.UpdateServerProfilesForV4(*server.ID, server.ProfileNames, tx); err != nil {
+		if err = dbhelpers.UpdateServerProfilesForV4(server.ID, server.Profiles, tx); err != nil {
 			userErr, sysErr, errCode := api.ParseDBError(err)
-			api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
-			return
+			return errCode, userErr, sysErr
 		}
 	} else {
-		if err = dbhelpers.UpdateServerProfileTableForV3(serverV3.ID, serverV3.ProfileID, (original.ProfileNames)[0], tx); err != nil {
-			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("failed to update server_profile: %w", err))
-			return
+		if err = dbhelpers.UpdateServerProfileTableForV3(serverV3.ID, serverV3.ProfileID, (original.Profiles)[0], tx); err != nil {
+			return http.StatusInternalServerError, nil, fmt.Errorf("failed to update server_profile: %w", err)
 		}
 	}
 
 	serverID, errCode, userErr, sysErr := updateServer(inf.Tx, server)
 	if userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
-		return
+		return errCode, userErr, sysErr
 	}
 
 	if userErr, sysErr, errCode = deleteInterfaces(id, tx); userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
-		return
+		return errCode, userErr, sysErr
 	}
 
 	if userErr, sysErr, errCode = createInterfaces(id, server.Interfaces, tx); userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
-		return
+		return errCode, userErr, sysErr
 	}
 
 	where := `WHERE s.id = $1`
 	var selquery string
-	if version.Major <= 4 {
+	if inf.Version.Major <= 4 {
```

Comment:
Instead of `LessThan`, `LessThanOrEqualTo`, `GreaterThan`, and `GreaterThanOrEqualTo`, we should just use `LessThan`, `GreaterThan` and logical inversions of those.

Reply:
`LessThan` is logically the same as `!GreaterThanOrEqualTo` and `LessThanOrEqualTo` is logically the same is `!GreaterThan`, but sometimes one representation is more readable than another, and being constrained to have to use `!` any time you want `OrEqualTo` makes the code less readable

Reply:
It's not a constraint, just a suggestion. You could just as easily use `v.GreaterThan(other) || v.Equal(other)`, or even `v.GreaterThan(other) || v == other`

Reply:
> It's not a constraint, just a suggestion. You could just as easily use `v.GreaterThan(other) || v.Equal(other)`, or even `v.GreaterThan(other) || v == other`.

The idea that the `<=` and `>=` operators are unnecessary is very silly. Some things are just more readable using a specific operator, and if that operator doesn't exist, the code is automatically less readable as a result.

> That will be confusing only if you force yourself to attempt to read it as "v is greater than or equal to other" rather than "v is not less than other".

Any code that forces everyone to think about it in the same way as the author is not good code and should be refactored

Reply:
You're equating `>=` and `<=` to `.GreaterThanOrEqualTo` and `LessThanOrEqualTo`, respectively. But those are literally different. These aren't operators, and they're just as equivalent to `.GreaterThan || .EqualTo` and `.LessThan || .EqualTo`. I don't care if the `OrEqualTo` versions exist, but it's crazy to me to say that, to extend your metaphor, having the pair of operators `>=` and `<` is somehow more sensible than just choosing the pair `>=` and `<=`, or `>` and `<`. There's no law that says we can't do all four, but if we're only going to have two I think it makes way more sense for them to be symmetric operations.

> Any code that forces everyone to think about it in the same way as the author is not good code and should be refactored

I couldn't agree less. Any code that leaves you with an inaccurate picture of the way the author thought about it is bad code and should be refactored. If I was stating these restrictions in English, "... an API version no less than 4.0..." seems more natural to me than "... an API version that is greater than or equal to 4.0...". If you think that's an uncommon way to phrase things, that's fair, but it's definitely because it's actually important to get the information across.